contexts/WalletManagerContext.tsx

```diff
@@ -660,10 +668,52 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
 
     fetchBalance();
 
-    rpcClient.addEventListener("utxos-changed", fetchBalance);
+    function checkIncomingUtxos(event: {
+      added: UtxoEntryReference[];
+      removed: UtxoEntryReference[];
+    }) {
+      if (event.added.length === 0) {
+        return;
+      }
+
+      const txId = event.added[0].outpoint.transactionId;
+      const network = settings?.networkId ?? NetworkType.Mainnet;
```

Comment:
We can use networkId from `useRpcClientStateful` instead of settings.

Reply:
I refactored the code to keep it consistent 782cd77db8ab302fbfe59691d2e34597414d4718